src/charm_types.py

```diff
@@ -20,3 +20,21 @@ class ExecResult(typing.NamedTuple):
     exit_code: int
     stdout: str
     stderr: str
+
+
+class DatasourcePostgreSQL(typing.TypedDict):
```

Comment:
I'd move the `ExecResult` class to synapse.py and rename this file to datasource.py or something similar. I'm not too keen on having a file for generic stuff